modules/build/src/main/scala/scala/build/preprocessing/ExtractedDirectives.scala

```diff
@@ -0,0 +1,107 @@
+package scala.build.preprocessing
+
+import com.virtuslab.using_directives.config.Settings
+import com.virtuslab.using_directives.custom.model.{
+  UsingDirectiveKind,
+  UsingDirectiveSyntax,
+  UsingDirectives
+}
+import com.virtuslab.using_directives.custom.utils.ast.{UsingDef, UsingDefs}
+import com.virtuslab.using_directives.{Context, UsingDirectivesProcessor}
+
+import scala.build.errors.{BuildException, DirectiveErrors}
+import scala.build.preprocessing.directives.StrictDirective
+import scala.build.{Logger, Position}
+import scala.jdk.CollectionConverters._
+
+case class ExtractedDirectives(
+  offset: Int,
+  directives: Seq[StrictDirective]
+)
+
+object ExtractedDirectives {
+
+  val changeToSpecialCommentMsg =
+    "Using directive using plain comments are deprecated. Please use a special comment syntax: '//> ...' or '/*> ... */'"
+
+  def from(
+    contentChars: Array[Char],
+    path: Either[String, os.Path],
+    logger: Logger,
+    supportedDirectives: Array[UsingDirectiveKind]
+  ): Either[BuildException, ExtractedDirectives] = {
+    val processor = {
+      val reporter = new DirectivesOutputStreamReporter(System.err) // TODO Get that via a logger
+      val settings = new Settings
+      settings.setAllowStartWithoutAt(true)
+      settings.setAllowRequire(false)
+      val context = new Context(reporter, settings)
+      new UsingDirectivesProcessor(context)
+    }
+    val all = processor.extract(contentChars, true, true).asScala
+
+    def byKind(kind: UsingDirectiveKind) = all.find(_.getKind == kind).get
+
+    def getDirectives(directives: UsingDirectives) =
+      directives.getAst() match {
+        case ud: UsingDefs =>
+          ud.getUsingDefs().asScala
+        case _ =>
+          Nil
+      }
+
+    val codeDirectives           = byKind(UsingDirectiveKind.Code)
+    val specialCommentDirectives = byKind(UsingDirectiveKind.SpecialComment)
+    val plainCommentDirectives   = byKind(UsingDirectiveKind.PlainComment)
+
+    def reportWarning(msg: String, values: Seq[UsingDef], before: Boolean = true): Unit =
+      values.foreach { v =>
+        val astPos = v.getPosition()
+        val (start, end) =
+          if (before) (0, astPos.getColumn())
+          else (astPos.getColumn(), astPos.getColumn() + v.getSyntax.getKeyword.size)
+        val position = Position.File(path, (astPos.getLine(), start), (astPos.getLine(), end))
+        logger.diagnostic(msg, positions = Seq(position))
+      }
+
+    val usedDirectives =
```

Comment:
In a situation when Java file starts with following code:

```
using foo 1
//> using bar 2
```

User will get a warning that bar is ignored and error that  foo is unsupported what is a bit confusing.

Reply:
Would you like to resolve priority only against supported directives?